### DISCUSSION TITLE
Copilot Chat - Excessive comments in generated code

### DISCUSSION THREAD

**Discussion by lostless13** [community/community#59697] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I find the excessive comments in code suggestions to be distracting and annoying. Copilot chat should strive to provide self-documenting code and I find it generally does. It would be nice if Copilot chat could be a bit more selective on when it adds comments or a configuration option to disable comments in responses. There is always the option to ask Copilot chat what a specific line does or block does.

Example response for some procedural code I had Copilot generate. All of these comments are redundant and unnecessary for all but the most inexperienced of programmers.
```
import os
from datetime import datetime, timedelta

def get_files(start_date, end_date):
    # Convert the date strings to datetime objects
    start_date = datetime.strptime(start_date, "%Y-%m-%d")
    end_date = datetime.strptime(end_date, "%Y-%m-%d")

    # Create an empty list to store the file paths
    file_paths = []

    # Loop through the date range and add the file paths to the list
    current_date = start_date
    while current_date <= end_date:
        # Format the date string
        date_str = current_date.strftime("%Y-%m-%d")

        # Create the file path
        file_path = f"{date_str}.csv.gz"

        # Check if the file exists
        if not os.path.exists(file_path):
            raise ValueError(f"File {file_path} does not exist")

        # Add the file path to the list
        file_paths.append(file_path)

        # Increment the date
        current_date += timedelta(days=1)

    # Return the list of file paths
    return file_paths
```

I find myself constantly asking Copilot to remove the comments so I can paste the code into my project.

**Comment by Akash1134** [comment]
Hi @lostless13 ,

Thanks for sharing the feedback. I have shared it with my team internally.

**Comment by travis-leith** [comment]
Comments are still excessive, 2 years later. I find it makes code harder to read when completely obvious functions, when clear names, have a comment that essentially mirrors the function name in a more verbose way.

**Comment by lostless13** [comment]
Agreed. Seen little improvement. Even with an instructions file that says not to include comments, it sometimes ignores the instructions and pukes unnecessary comments all over the code. Most modern code has very few comments. I don't know where it is getting this behaviour from.
